packages/core/src/node-execution-context/__tests__/base-context.test.ts

```diff
@@ -0,0 +1,168 @@
+import { mock } from 'jest-mock-extended';
+import type {
+	INode,
+	INodeExecutionData,
+	IWorkflowExecuteAdditionalData,
+	Workflow,
+	WorkflowExecuteMode,
+} from 'n8n-workflow';
+import { Container } from 'typedi';
+
+import { InstanceSettings } from '@/InstanceSettings';
+
+import { BaseContext } from '../base-contexts';
+
+class TestContext extends BaseContext {}
+
+describe('BaseContext', () => {
+	const instanceSettings = mock<InstanceSettings>({ instanceId: 'abc123' });
+	Container.set(InstanceSettings, instanceSettings);
+
+	const workflow = mock<Workflow>({
+		id: '123',
+		name: 'Test Workflow',
+		active: true,
+		nodeTypes: mock(),
+		timezone: 'UTC',
+	});
+	const node = mock<INode>();
+	let additionalData = mock<IWorkflowExecuteAdditionalData>({
+		credentialsHelper: mock(),
+	});
+
+	const mode: WorkflowExecuteMode = 'manual';
+	const testContext = new TestContext(workflow, node, additionalData, mode);
+
+	beforeEach(() => {
+		jest.clearAllMocks();
+	});
+
+	describe('getNode', () => {
+		it('should return a deep copy of the node', () => {
+			const result = testContext.getNode();
+			expect(result).not.toBe(node);
+			expect(JSON.stringify(result)).toEqual(JSON.stringify(node));
```

Comment:
for some weird reason `result` when logged to console is correct, but on `.toEqual` it's `undefined`. I couldn't wrap my head around what was happening because of all the proxy mock objects. I can clean this up later.

Reply:
Not a biggie, this is completely fine as well 👍